File: sdi3-5.Cli-EJB/src/uo/sdi/client/DisableUser.java
package uo.sdi.client;

import uo.sdi.business.AdminService;
import uo.sdi.business.exception.BusinessException;
import uo.sdi.business.impl.RemoteEJBServicesLocator;
import uo.sdi.dto.User;
import uo.sdi.dto.types.UserStatus;
import alb.util.console.Console;
import alb.util.menu.Action;

public class DisableUser implements Action{

	@Override
	public void execute() throws Exception {
		AdminService service = new RemoteEJBServicesLocator().getAdminService();
		long id = Console.readLong("Enter an id: ");	
		try{
			User user = service.findUserById(id);
			if (user==null){
				System.out.println("There is no user with id:"+id);
				return;
			}
				
			if(user.getStatus().equals(UserStatus.ENABLED)){
				service.disableUser(id);
				System.out.println("The user with id "+id+" has been "
						+ "successfully disabled");
			}
			else{
				service.enableUser(id);
				System.out.println("The user with id "+id+" has been "
						+ "successfully enabled");
			}	
		} catch(BusinessException b){
			System.out.println("The user could not be disabled due to: \n"
					+b.getLocalizedMessage());
		}
	}

}

File: sdi3-5.Cli-SOAP-C_SHARP/sdi3-5.Cli-SOAP-C_SHARP/DisableUser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sdi3_5.Cli_SOAP_C_SHARP
{
    class DisableUser : Command
    {
        public void Execute()
        {
                AdminService.AdminService service = new AdminService.AdminServiceClient();

                long id = Util.ReadLong("Enter an id",0);

                AdminService.findUserById requestUser = new AdminService.findUserById();
                requestUser.arg0 = id;
                AdminService.user user = service.findUserById(requestUser).@return;

                if (user == null)
                {
                    Console.WriteLine("There is no user with " + "id " + id);
                    return;
                }
                
                   
  
                
                if (user.status.Equals(AdminService.userStatus.ENABLED))
                {
                    AdminService.disableUser requestDisable = new AdminService.disableUser();
                    requestDisable.arg0 = id;
                    service.disableUser(requestDisable);
                   Console.WriteLine("The user with id " + id + " has been "
                            + "successfully disabled");
                }
                else
                {
                        AdminService.enableUser requestEnable = new AdminService.enableUser();
                        requestEnable.arg0 = id;
                        service.enableUser(requestEnable);
                        Console.WriteLine("The user with id " + id + " has been "
                            + "successfully enabled");
                }
            
        }
    }
}
